src/Orleans/Runtime/GrainReference.cs

```diff
@@ -407,6 +407,8 @@ private static void ResponseCallback(Message message, TaskCompletionSource<objec
             {
                 context.TrySetException(response.Exception);
             }
+
+            message.Dispose();
```

Comment:
" One or few leaked messages are nothing" - I don't completely disagree, but we should be mindful that leaking pooled messages is significantly different from leaking regular short lived messages.  If a message is allocated and freed, that's quite cheap.  If it's allocated and stays around long enough to be compacted in gen2, then freeing it is significantly heavier.  In general it's better to allocate and free, than allocate, hold, then free.

Having said that, I suspect your probably correct, odds of this code throwing looks low.